agent/hcp/manager.go

```diff
@@ -96,6 +95,13 @@ func (m *Manager) Run(ctx context.Context) {
 			"error", err)
 	}
 
+	// Update the telemetry provider to enable the HCP metrics sink
+	if err := m.updateTelemetryProvider(); err != nil {
```

Comment:
Just adding a comment here based on our sync discussion.

I was wondering how we would dynamically update the telemetry config later, but turns out by this point both scada/telemetry should be configurable. I think we should actually handle this error. 

Some design here is more futuristic, and we can always bring back this idea of updating in the future!

Reply:
Related: if so, did we consider just starting the TelemetryProvider in here? passing in the config once available. I might be missing some context / forgetting something here.

Reply:
I thought that the metrics sink wouldn't work properly if the telemetry provider wasn't started, but since that doesn't seem to be the case (defaults to a disabled config), then I think that would be an improvement! That way we don't have the process running unless needed.

To clarify, are you also suggesting that we pass the configuration values as part of the `Run` call?

Reply:
@mkam yep pass in the initial runtime/http config to the run when starting it. Yeah, the default config would just disable the sink, so it should be okay I think. Might be good to test a case without runtime config E2E just to make sure 💯

Reply:
Made these changes! Here's the results of testing it again too:

Manual inspection of logs:
```
2024-01-11T13:47:18.551-0600 [DEBUG] agent.hcp: starting telemetry config provider
2024-01-11T13:47:18.551-0600 [TRACE] agent.hcp.telemetry_config_provider: fetching telemetry config
2024-01-11T13:47:18.977-0600 [TRACE] agent.hcp.telemetry_config_provider: successfully fetched telemetry config
2024-01-11T13:48:18.373-0600 [DEBUG] agent.hcp.hcp_telemetry_client: performing request: method=POST url=https://consul-telemetry.cloud.hashicorp.com/otlp/v1/metrics
2024-01-11T13:49:18.372-0600 [DEBUG] agent.hcp.hcp_telemetry_client: performing request: method=POST url=https://consul-telemetry.cloud.hashicorp.com/otlp/v1/metrics
2024-01-11T13:50:18.371-0600 [DEBUG] agent.hcp.hcp_telemetry_client: performing request: method=POST url=https://consul-telemetry.cloud.hashicorp.com/otlp/v1/metrics
2024-01-11T13:51:18.373-0600 [DEBUG] agent.hcp.hcp_telemetry_client: performing request: method=POST url=https://consul-telemetry.cloud.hashicorp.com/otlp/v1/metrics
2024-01-11T13:52:18.373-0600 [DEBUG] agent.hcp.hcp_telemetry_client: performing request: method=POST url=https://consul-telemetry.cloud.hashicorp.com/otlp/v1/metrics
2024-01-11T13:52:18.981-0600 [TRACE] agent.hcp.telemetry_config_provider: fetching telemetry config
2024-01-11T13:52:19.282-0600 [TRACE] agent.hcp.telemetry_config_provider: successfully fetched telemetry config
2024-01-11T13:53:18.374-0600 [DEBUG] agent.hcp.hcp_telemetry_client: performing request: method=POST url=https://consul-telemetry.cloud.hashicorp.com/otlp/v1/metrics
2024-01-11T13:54:18.376-0600 [DEBUG] agent.hcp.hcp_telemetry_client: performing request: method=POST url=https://consul-telemetry.cloud.hashicorp.com/otlp/v1/metrics
2024-01-11T13:55:18.375-0600 [DEBUG] agent.hcp.hcp_telemetry_client: performing request: method=POST url=https://consul-telemetry.cloud.hashicorp.com/otlp/v1/metrics
```

Running consul-k8s tests:
```
--- PASS: TestObservabilityCloud (349.85s)
    --- PASS: TestObservabilityCloud/default_namespace_and_partition (236.82s)
        --- PASS: TestObservabilityCloud/default_namespace_and_partition/collectorExportsMetrics (94.39s)
        --- PASS: TestObservabilityCloud/default_namespace_and_partition/consulPeriodicRefreshUpdateConfig (33.14s)
        --- PASS: TestObservabilityCloud/default_namespace_and_partition/consulPeriodicRefreshDisabled (0.01s)
    --- PASS: TestObservabilityCloud/default_namespace_and_partition;_secure (113.03s)
        --- SKIP: TestObservabilityCloud/default_namespace_and_partition;_secure/collectorExportsMetrics (0.00s)
        --- SKIP: TestObservabilityCloud/default_namespace_and_partition;_secure/consulPeriodicRefreshUpdateConfig (0.00s)
        --- SKIP: TestObservabilityCloud/default_namespace_and_partition;_secure/consulPeriodicRefreshDisabled (0.00s)
    --- SKIP: TestObservabilityCloud/namespace_mirroring;_secure (0.00s)
    --- SKIP: TestObservabilityCloud/admin_partitions;_secure (0.00s)
PASS
ok  	github.com/hashicorp/consul-k8s/acceptance/tests/cloud	350.675s

```

Reply:
Looks like this change caused a race condition, will investigate!

Reply:
Fixed the race condition and reran the tests! 😅